File: 14-longest-common-prefix/14-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string pre=strs[0];
        for(int i=1;i<strs.size();i++)
        {
            int j=0;
            while(j<strs[i].size())
            {
                if(pre[j]==strs[i][j])
                    j++;
                else break;
            }
            pre.erase(pre.begin()+j,pre.end());
        }
        return pre;
    }
};

File: 14-longest-common-prefix/14-longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    p := strs[0]
    for _, s := range strs {
        i := 0
        for ; i < len(s) && i < len(p) && p[i] == s[i]; i++ {}
        p = p[:i]
    }
    return p
}